ruby/neo4j/driver/internal/cluster/rediscovery_impl.rb

```diff
@@ -49,6 +49,7 @@ def resolve
             resolve_all_by_domain_name(server_address).unicast_stream
             # rescue java.net.UnknownHostException => e
           rescue => e
+            e = e.is_a?(SocketError) ? Exceptions::SocketException.new : e
             exception&.add_suppressed(e)
```

Comment:
```suggestion
          rescue SocketError => e
```
let's just store the 1st exception. There is not much info in those exceptions anyway.